packages/core/echo/echo-db/src/query/query-state.ts

```diff
@@ -163,7 +163,7 @@ export class QueryState extends Resource {
 // TODO(burdon): Process Filter DSL.
 const filterToIndexQuery = (filter: Filter): IndexQuery => {
   return {
-    typename: filter.type?.itemId,
+    typenames: filter.type?.itemId ? [filter.type.itemId] : filter.or.map((f) => f.type?.itemId).filter(nonNullable),
```

Comment:
what if some of the nested filters have a .not flag?

Reply:
I guess something (this function? query state?) should bail and not do an indexed search if the index doesn't support this type of query